rmqsrc/connections.nim

```diff
@@ -278,18 +299,67 @@ proc onDataAvailable*(connection: Connection, data: string) =
     except IOError:
       return
 
+proc onConnectionStart(connection: Connection, methodFrame: Frame) =
+  connection.log "Starting AMQP connection."
+  connection.state = csStart
+  connection.serverProperties = methodFrame.rpcMethod.mStartParams.serverProperties
+  # if self._is_protocol_header_frame(method_frame):
+  #     raise exceptions.UnexpectedFrameError
+  # self._check_for_protocol_mismatch(method_frame)
+  # self._set_server_information(method_frame)
+
+  # TODO parametrize connection start response string?
+  connection.sendConnectionStartOk(methodFrame, connection.getCredentials(methodFrame))
+
+proc onConnectionTune(connection: Connection, methodFrame: Frame) =
+  let frameParams = methodFrame.rpcMethod.mTuneParams
+  connection.log "Starting AMQP tuning."
+  connection.state = csTune
+  let
+    channelMax = negotiateIntegerValue(
+      connection.parameters.channelMax, frameParams.channelMax
+    )
+    frameMax = negotiateIntegerValue(
+      connection.parameters.frameMax, frameParams.frameMax
+    )
+    heartbeat = negotiateIntegerValue(
+      connection.parameters.heartbeat, frameParams.heartbeat
+    )
```

Comment:
There are some subtleties we can fill in, but basically it takes the server and client tuning values and says: for each value -> if one is 0, use the other one; otherwise, use the smaller one.